diff --git a/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm b/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm
index 01fa493..7404dc1 100644
--- a/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm
+++ b/node_modules/lottie-react-native/ios/Fabric/LottieAnimationViewComponentView.mm
@@ -19,6 +19,10 @@ @implementation LottieAnimationViewComponentView {
     LottieContainerView *_view;
 }
 
++(void) load {
+    [super load];
+}
+
 - (instancetype) initWithFrame:(CGRect)frame
 {
     if (self = [super initWithFrame:frame]) {
diff --git a/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift b/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift
index f4f575d..8c9a82e 100644
--- a/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift
+++ b/node_modules/lottie-react-native/ios/LottieReactNative/ContainerView.swift
@@ -286,11 +286,9 @@ class ContainerView: RCTView {
         play()
     }
 
-    // The animation view is a child of the RCTView, so if the bounds ever change, add those changes to the animation view as well
-    override var bounds: CGRect {
-        didSet {
-            animationView?.frame = self.bounds
-        }
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        animationView?.frame = self.bounds
     }
 
     // MARK: Private
diff --git a/node_modules/lottie-react-native/lottie-react-native.podspec b/node_modules/lottie-react-native/lottie-react-native.podspec
index ea3496a..3150a4f 100644
--- a/node_modules/lottie-react-native/lottie-react-native.podspec
+++ b/node_modules/lottie-react-native/lottie-react-native.podspec
@@ -2,7 +2,17 @@ require 'json'
 
 package = JSON.parse(File.read(File.join(__dir__, "package.json")))
 
-folly_compiler_flags = '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32'
+def self.is_install_modules_dependencies_needed
+  begin
+    # Read the React Native version from the package.json relative to the podspec file
+    react_package_file = File.expand_path("../react-native/package.json", File.dirname(__FILE__))
+    react_package = JSON.parse(File.read(react_package_file))
+    minor_react_version = react_package['version'].split('.')[1].to_i
+    return minor_react_version >= 73 && defined?(install_modules_dependencies)
+  rescue
+    return true
+  end
+end
 
 Pod::Spec.new do |s|
   s.name                    = package["name"]
@@ -24,23 +34,35 @@ Pod::Spec.new do |s|
   
   s.swift_version = '5.6'
 
-  if ENV['RCT_NEW_ARCH_ENABLED'] == '1' then
-    s.compiler_flags = folly_compiler_flags + " -DRCT_NEW_ARCH_ENABLED=1"
-    s.pod_target_xcconfig    = {
-      "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/boost\"",
-      "OTHER_CPLUSPLUSFLAGS" => "-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1",
-      "CLANG_CXX_LANGUAGE_STANDARD" => "c++17",
-      "DEFINES_MODULE" => "YES",
-    }
-    
-    s.dependency "React-RCTFabric"
-    s.dependency "React-Codegen"
-    s.dependency "RCT-Folly"
-    s.dependency "RCTRequired"
-    s.dependency "RCTTypeSafety"
-    s.dependency "ReactCommon/turbomodule/core"
+  # TODO(workaround): checking RN version for backward comaptibility reasons since lottie is not compatible with RN 72 and below with install_modules_dependencies
+  # TODO: We need to replace this line with   if defined?(install_modules_dependencies)
+  if is_install_modules_dependencies_needed then
+    print "Using install_modules_dependencies\n"
+    install_modules_dependencies(s)
+    if !(ENV['RCT_NEW_ARCH_ENABLED'] == '1') then
+      s.exclude_files = "ios/Fabric"
+    end
   else
-    s.dependency 'React-Core'
-    s.exclude_files = "ios/Fabric"
+    print "Not using install_modules_dependencies\n"
+    if ENV['RCT_NEW_ARCH_ENABLED'] == '1' then
+      folly_compiler_flags = '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32'
+      s.compiler_flags = folly_compiler_flags + " -DRCT_NEW_ARCH_ENABLED=1"
+      s.pod_target_xcconfig    = {
+        "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/boost\"",
+        "OTHER_CPLUSPLUSFLAGS" => "-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1",
+        "CLANG_CXX_LANGUAGE_STANDARD" => "c++17",
+        "DEFINES_MODULE" => "YES",
+      }
+
+      s.dependency "React-RCTFabric"
+      s.dependency "React-Codegen"
+      s.dependency "RCT-Folly"
+      s.dependency "RCTRequired"
+      s.dependency "RCTTypeSafety"
+      s.dependency "ReactCommon/turbomodule/core"
+    else
+      s.dependency 'React-Core'
+      s.exclude_files = "ios/Fabric"
+    end
   end
 end
